pkg/scale/encode.go

```diff
@@ -56,32 +57,68 @@ func (es *encodeState) marshal(in interface{}) (err error) {
 	switch in := in.(type) {
 	case int:
 		err = es.encodeUint(uint(in))
+		if err != nil {
+			err = fmt.Errorf("encoding Uint: %w", err)
+		}
 	case uint:
 		err = es.encodeUint(in)
+		if err != nil {
+			err = fmt.Errorf("encoding Uint: %w", err)
+		}
 	case int8, uint8, int16, uint16, int32, uint32, int64, uint64:
 		err = es.encodeFixedWidthInt(in)
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case *big.Int:
 		err = es.encodeBigInt(in)
+		if err != nil {
+			err = fmt.Errorf("encoding BigInt: %w", err)
+		}
 	case *Uint128:
 		err = es.encodeUint128(in)
+		if err != nil {
+			err = fmt.Errorf("encoding Uint128: %w", err)
+		}
 	case []byte:
 		err = es.encodeBytes(in)
+		if err != nil {
+			err = fmt.Errorf("encoding []byte: %w", err)
```

Comment:
supernit just that when I read `[]byte` I thought it was the actual value. Better write in human focused strings in this case 😉 


```suggestion
			err = fmt.Errorf("encoding byte slice: %w", err)
```